src/catch2/catch_totals.cpp

```diff
@@ -14,21 +14,23 @@ namespace Catch {
         diff.passed = passed - other.passed;
         diff.failed = failed - other.failed;
         diff.failedButOk = failedButOk - other.failedButOk;
+        diff.skipped = skipped - other.skipped;
         return diff;
     }
 
     Counts& Counts::operator += ( Counts const& other ) {
         passed += other.passed;
         failed += other.failed;
         failedButOk += other.failedButOk;
+        skipped += other.skipped;
         return *this;
     }
 
     std::uint64_t Counts::total() const {
-        return passed + failed + failedButOk;
+        return passed + failed + failedButOk + skipped;
     }
     bool Counts::allPassed() const {
-        return failed == 0 && failedButOk == 0;
+        return failed == 0 && failedButOk == 0 && skipped == 0;
```

Comment:
As you say in the PR description, whether skipped tests should count as passing or not is a good question. IIRC GTest counts them as passing, so skipping tests does not cause non-zero return code.

I think this is what the users will expect. but this still leaves one open question:

What should happen when _all_ tests were skipped?  The current behaviour of Catch2 is that if no tests were run, then the return code is non-zero, even though no tests have failed. What if two tests were selected to be run, but they were both skipped?

I think that the answer should be that if all selected tests were skipped, then the return code should also be non-zero. Obviously this will need new tests 😃

Reply:
I've kept `allPassed` to only be true when no tests are being skipped. This does not affect the exit code; in fact this function is only used in three locations across two reporters. If adding skipped assertions/tests into the mix is muddying the semantics too much, I suggest to simply remove `Counts::allPassed` altogether.